src/posix/platform/udp.cpp

```diff
@@ -427,6 +435,94 @@ otError otPlatUdpSend(otUdpSocket *aUdpSocket, otMessage *aMessage, const otMess
     return error;
 }
 
+otError otPlatUdpJoinMulticastGroup(otUdpSocket *       aUdpSocket,
+                                    otNetifIdentifier   aNetifIdentifier,
+                                    const otIp6Address *aAddress)
+{
+    otError          error = OT_ERROR_NONE;
+    struct ipv6_mreq mreq;
+    int              fd;
+
+    VerifyOrExit(aUdpSocket->mHandle != nullptr, error = OT_ERROR_INVALID_ARGS);
+    fd = FdFromHandle(aUdpSocket->mHandle);
+
+    memcpy(&mreq.ipv6mr_multiaddr, aAddress->mFields.m8, sizeof(mreq.ipv6mr_multiaddr));
+
+    switch (aNetifIdentifier)
+    {
+    case OT_NETIF_UNSPECIFIED:
+        break;
+    case OT_NETIF_THREAD:
+        mreq.ipv6mr_interface = gNetifIndex;
+        break;
+    case OT_NETIF_BACKBONE:
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+        mreq.ipv6mr_interface = gBackboneNetifIndex;
+#else
+        ExitNow(error = OT_ERROR_NOT_IMPLEMENTED);
+#endif
+        break;
+    }
+
+    VerifyOrExit(setsockopt(fd, IPPROTO_IPV6, IPV6_JOIN_GROUP, &mreq, sizeof(mreq)) != 0);
+
+    if (errno == EADDRINUSE)
+    {
+        error = OT_ERROR_NONE;
+    }
```

Comment:
is it necessary?

Reply:
Yes.  `errno == EADDRINUSE` means the socket is already in the `JOIN_GROUP` and we are joining it once again. I think we should treat such case as successful. 

Currently, BTMF can subscribe the `All Network BBRs` multicast address for multiple time since `BackboneTmfAgent::Start` might be called when it's already started.